chronos/internal/asyncfutures.nim

```diff
@@ -1515,3 +1553,53 @@ when defined(windows):
 
     retFuture.cancelCallback = cancellation
     return retFuture
+
+{.pop.} # Automatically deduced raises from here onwards
+
+template fail*[T, E](
+    future: InternalRaisesFuture[T, E], error: ref CatchableError,
+    warn: static bool = true) =
+  checkFailureType(future, error, warn)
+  fail(future, error, getSrcLocation())
+
+proc waitFor*[T, E](fut: InternalRaisesFuture[T, E]): T = # {.raises: [E]}
+  ## **Blocks** the current thread until the specified future completes.
+  ## There's no way to tell if poll or read raised the exception
```

Comment:
This comment is ugly one, `poll()` do not raise exceptions anymore.

Reply:
174644ef149ba6a40d05975cc4821bdfbb328735